src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -706,14 +697,12 @@ public Visitable visit(ObjectCreationExpr newExpr, Void p) {
     try {
       List<String> argumentsCreation = getArgumentsFromObjectCreation(newExpr);
       UnsolvedMethod creationMethod = new UnsolvedMethod("", type, argumentsCreation);
-      if (classAndPackageMap.containsKey(type)) {
-        UnsolvedClass newClass = new UnsolvedClass(type, classAndPackageMap.get(type));
-        newClass.addMethod(creationMethod);
-        this.updateMissingClass(newClass);
-      } else {
-        throw new RuntimeException("Unexpected class: " + type);
-      }
-
+      // we assume that an unsolved class not found among import statements should be in the same
+      // package as the current class
+      String packageName = classAndPackageMap.getOrDefault(type, currentPackage);
+      UnsolvedClass newClass = new UnsolvedClass(type, packageName);
+      newClass.addMethod(creationMethod);
```

Comment:
as we've added `updateMissingClass` to the `updateUnsolvedClass` method, I think we can't call `updateUnsolvedClass` here because we need to add the method to the `UnsolvedClass` instance before updating the list of missing classes.

Reply:
> we need to add the method to the UnsolvedClass instance before updating the list of missing classes

Why? The pointer is still valid, and I don't think there's any reason to believe that any computation will be done with the list of missing classes before the call to `addMethod` is executed (assuming a sequential Java semantics, but I'm okay with that assumption here given the way the JVM works).

Reply:
No, I am wrong: `updateMissingClass` does more computation than I expected. I was incorrectly thinking that it just updates a data structure; it also does computation over the list of methods.

Can you modify `updateUnsolvedClass` to take a varargs array of methods and add them all to the class before it calls `updateMissingClass`?

Reply:
Professor, I'm afraid that might make `updateUnsolvedClass` more complicated than it should be. The reason is that it will require an additional array as an argument whenever it is called, even if there is no methods added (such as in line 529 and 674). And even at line 704, while there is a method added, there is only one method added. So modifying `updateUnsolvedClass` to take an array of methods as an argument seems to be unnecessary.

Reply:
> it will require an additional array as an argument whenever it is called

It will not, if you use the [varargs](https://docs.oracle.com/javase/8/docs/technotes/guides/language/varargs.html) feature as I suggested. It is designed for exactly this situation, and the resulting code should be simpler than what you currently have.